commands/verify/verify.go

```diff
@@ -96,44 +154,75 @@ func DoVerifyAccordingConfig() error {
 	e2eConfig := config.GlobalConfig.E2EConfig
 
 	retryCount := e2eConfig.Verify.RetryStrategy.Count
-	if retryCount <= 0 {
-		retryCount = 1
-	}
+	retryCount = checkForRetryCount(retryCount)
+
 	interval, err := parseInterval(e2eConfig.Verify.RetryStrategy.Interval)
 	if err != nil {
 		return err
 	}
 
 	failFast := e2eConfig.Verify.FailFast
+	concurrency := e2eConfig.Verify.Concurrency
+
+	var errCollection multierror.Error
+
+	errContain := ErrContain{
+		errAddr: &errCollection,
+	}
 
-	var errCollection *multierror.Error
-	for idx, v := range e2eConfig.Verify.Cases {
-		if v.GetExpected() == "" {
-			errMsg := fmt.Sprintf("the expected data file for case[%v] is not specified\n", idx)
-			if failFast {
-				return errors.New(errMsg)
+	if concurrency {
+		var waitGroup sync.WaitGroup
+		VerifyInfo := verifyInfo{
+			retryCount,
+			interval,
+			failFast,
+		}
+		chanBool := make(chan bool, 1)
+		waitGroup.Add(len(e2eConfig.Verify.Cases))
+
+		for idx, v := range e2eConfig.Verify.Cases {
+			go concurrentVerifySingleCase(idx, v, &errContain, VerifyInfo, &waitGroup, chanBool)
+		}
+
+		if failFast {
+			for {
+				result := <-chanBool
+				if result {
+					return errContain.errAddr.ErrorOrNil()
+				}
 			}
-			logger.Log.Warnf(errMsg)
-			errCollection = multierror.Append(errCollection, errors.New(errMsg))
-			continue
 		}
 
-		for current := 1; current <= retryCount; current++ {
-			if err := verifySingleCase(v.GetExpected(), v.GetActual(), v.Query); err == nil {
-				break
-			} else if current != retryCount {
-				logger.Log.Warnf("verify case failure, will continue retry, %v", err)
-				time.Sleep(interval)
-			} else {
+		waitGroup.Wait()
+	} else {
+		for idx, v := range e2eConfig.Verify.Cases {
+			if v.GetExpected() == "" {
+				errMsg := fmt.Sprintf("the expected data file for case[%v] is not specified\n", idx)
 				if failFast {
-					return err
+					return errors.New(errMsg)
+				}
+				logger.Log.Warnf(errMsg)
+				errContain.errAddr = multierror.Append(errContain.errAddr, errors.New(errMsg))
+				continue
+			}
+
+			for current := 1; current <= retryCount; current++ {
+				if err := verifySingleCase(v.GetExpected(), v.GetActual(), v.Query); err == nil {
+					break
+				} else if current != retryCount {
+					logger.Log.Warnf("verify case failure, will continue retry, %v", err)
+					time.Sleep(interval)
+				} else {
+					if failFast {
+						return err
+					}
+					errContain.errAddr = multierror.Append(errContain.errAddr, err)
 				}
-				errCollection = multierror.Append(errCollection, err)
 			}
 		}
 	}
 
-	return errCollection.ErrorOrNil()
+	return errContain.errAddr.ErrorOrNil()
```

Comment:
Since we passed the pointer of `errCollection` to `errContain`, no need to modify this.